src/reflect/scala/reflect/internal/util/Parallel.scala

```diff
@@ -0,0 +1,85 @@
+package scala.reflect.internal.util
+
+import java.util.concurrent.atomic.AtomicInteger
+
+object Parallel {
+
+  class WorkerThread(group: ThreadGroup, target: Runnable, name: String,
```

Comment:
we need a SingleThreadedWorker as dicussed on call

Reply:
Running in the current thread is handled inside execution context in `Global`:

```scala
private def createExecutionContext(): ExecutionContextExecutor = {
      if (settings.YparallelPhases.containsPhase(this)) {
        val parallelThreads = settings.YparallelThreads.value
        val threadPoolFactory = ThreadPoolFactory(Global.this, this)
        val javaExecutor = threadPoolFactory.newUnboundedQueueFixedThreadPool(parallelThreads, "worker")
        ExecutionContext.fromExecutorService(javaExecutor, _ => ())
      } else ExecutionContext.fromExecutor((task: Runnable) => asWorkerThread(task.run()))
    }
```